### DISCUSSION TITLE
on_offset with a dynamic offset value

### DISCUSSION THREAD

**Discussion by camilocerchiari** [savonet/liquidsoap#3345] [Q&A]
hi!
I'm looking for a way to call a handler funtion on a given offset of the currently playing source, with the catch that the offset will chage from time to time.
I tryed using [on_offset](https://www.liquidsoap.info/doc-2.2.0/reference.html#on_offset) for this and it works great if the offset is a constant.  then I tryed passing a ref instead of the constant, whose value changes, but `on_offset` seems to have evaluated it the first time and is keeping that first value.  same thing if I use a function instead.

what I want is something like a lazy evaluation of the parameters.
is there any way to achieve this?

**Comment by smimram** [comment]
Hi, I haved just allowed getters for the `offset` in https://github.com/savonet/liquidsoap/pull/3355.

**Reply by camilocerchiari** [reply]
awesome, thank you!
